### DISCUSSION TITLE
Spoiler-tag/blur option for uploading and linking media

### DISCUSSION THREAD

**Discussion by Miacelium** [element-hq/element-meta#1115] [Ideas]
#### What would you like to do?
I'd like for people to have an easy time adding spoiler/content-warnings to posts.

#### Why would you like to do it?
It'd let people opt-in to seeing potentially triggering content or simply hide media spoilers when discussing things like TV Shows and Movies.

#### How would you like to achieve it?
Since I migrated from Discord and most of my friends too, I'm used to having a toggle when uploading media directly and tags for links and media embedded through links. Discord adds a "SPOILER_" to the mark them to be blurred, when uploading media directly. It'd be very helpful if on top it'd then have the option to add a description/reason for why the uploader decided it might need to be blurred.

#### Have you considered any alternatives?
As far as I understood in other threads /spoiler exists but can't censor media/links and you'd have to use HTML tags(?)

**Comment by CrimsonFork** [comment]
It's imperfect but I'm working on a [tool](https://codeberg.org/crimsonfork/spoilerinator) that partially allows to get there, although it's limited due to the inconsistent implementation of spoilers across clients.
